### PR TITLE
[SPARK-42656][FOLLOWUP] Add BUILD and SCCLASSPATH options to Spark Connect scripts

### PR DESCRIPTION
**Pull request by juliuszsompolski**
### What changes were proposed in this pull request?

By default, scripts in `connector/connect/bin` rebuild the necessary JARs using SBT. Even when this build is incremental and there were no changes, the sbt rebuild takes about 4 minutes. Another issue is that launching the `connector/connect/bin/spark-connect`, `connector/connect/bin/spark-connect-shell`, `connector/connect/bin/spark-connect-scala-client` scripts in parallel causes the two sbt builds to clash. So developer has to wait for one to finish before launching the other.

By supplying an env var option `BUILD=0` the build step can be skipped:
```
BUILD=0 ./connector/connect/bin/spark-connect
BUILD=0 ./connector/connect/bin/spark-connect-shell
BUILD=0 ./connector/connect/bin/spark-connect-scala-client
```
At the same time, add a new script `connector/connect/bin/spark-connect-build` to rebuild both the server and client part:
```
./connector/connect/bin/spark-connect-build
```

After doing that, it's still slow to have sbt retrieve the classpath of the scala client at each startup. With `connector/connect/bin/spark-connect-scala-client-classpath` and new `SCCLASSPATH` env argument, it's possible to save it to a file:
```
connector/connect/bin/spark-connect-scala-client-classpath > ./scclasspath
BUILD=0 SCCLASSPATH=$(cat ./scclasspath) connector/connect/bin/spark-connect-scala-client
```

### Why are the changes needed?

Improving inner dev loop.

### Does this PR introduce _any_ user-facing change?

No.

### How was this patch tested?

Manual build, using the scripts.

### CODE REVIEW

**Comment by juliuszsompolski** [comment]
@zhenlineo @hvanhovell

**Comment by juliuszsompolski** [comment]
The client script still does
```
CONNECT_CLASSPATH="$(build/sbt "${SCALA_ARG}" -DcopyDependencies=false "export connect-client-jvm/fullClasspath" | grep jar | tail -n1)"
SQL_CLASSPATH="$(build/sbt "${SCALA_ARG}" -DcopyDependencies=false "export sql/fullClasspath" | grep jar | tail -n1)"
```
and these sbt commands still take almost a minute (20-30s each) before it drops me into the client shell...
I wonder if something is wrong with my setup.

**Review by zhenlineo — APPROVED** [review]
LGTM. Can you update the new dev commands in the PR description too?

**Comment by juliuszsompolski** [comment]
Updated PR description.

**Comment by juliuszsompolski** [comment]
Also added possibility to provide saved classpath without waiting 1 minute for sbt to compute it.

**Review by hvanhovell — APPROVED** [review]
LGTM

File: connector/connect/bin/spark-connect [hunk 1]
```diff
@@ -30,8 +30,11 @@ export SPARK_HOME=$FWDIR
 SCALA_BINARY_VER=`grep "scala.binary.version" "${SPARK_HOME}/pom.xml" | head -n1 | awk -F '[<>]' '{print $3}'`
 SCALA_ARG="-Pscala-${SCALA_BINARY_VER}"
 
-# Build the jars needed for spark submit and spark connect
-build/sbt "${SCALA_ARG}" -Phive -Pconnect package
+BUILD="${BUILD:-1}"
```
Comment (HyukjinKwon) [inline connector/connect/bin/spark-connect:33]: Sorry for a posthoc review. I wonder if we should better pick a different name though. `BUILD` might be too common ..  and that might easily conflict.

Comment (juliuszsompolski) [reply connector/connect/bin/spark-connect:33]: Ha, that's exactly why I did SCCLASSPATH, but didn't think about this. Let me change it to SCBUILD in a followup.

**Comment by HyukjinKwon** [comment]
Hm .. this only got into the master branch but the original fix (https://github.com/apache/spark/pull/40257). Would probably need a separate JIRA in such case next time ;-).

**Comment by juliuszsompolski** [comment]
I think it's fine in master only. It's dev scripts and probably majority of dev will be in master moving forward.

**Comment by HyukjinKwon** [comment]
Yeah, merging it to master only is fine. The point I was making was to have a different JIRA so we know which version the fix has landed to (so as to make it easier when we revert, etc.)

**Pull request closed without merging**